backend/service/github/github.go

```diff
@@ -51,9 +51,11 @@ type StatsRoundTripper struct {
 func (st *StatsRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	resp, err := st.Wrapped.RoundTrip(req)
 
-	if hdr := resp.Header.Get("X-RateLimit-Remaining"); hdr != "" {
-		if v, err := strconv.Atoi(hdr); err == nil {
-			st.scope.Gauge("rate_limit_remaining").Update(float64(v))
+	if resp != nil {
```

Comment:
hmm i wonder if we should be checking for `if err != nil` instead?

from the package code comments it says

```
	// RoundTrip should not attempt to interpret the response. In
	// particular, RoundTrip must return err == nil if it obtained
	// a response, regardless of the response's HTTP status code.
	// A non-nil err should be reserved for failure to obtain a
	// response. Similarly, RoundTrip should not attempt to
	// handle higher-level protocol details such as redirects,
	// authentication, or cookies.
```

so a non nil error would let us know if there was an error getting a response

Reply:
yeah I originally had the check for error being non-nil but I realized it makes an assumption that resp is always nil when err is set when under the hood we rely on resp